test/fetch/encoding.js

```diff
@@ -17,10 +17,10 @@ test('content-encoding header is case-iNsENsITIve', async (t) => {
     res.setHeader('Content-Encoding', contentCodings)
     res.setHeader('Content-Type', 'text/plain')
 
-    gzip.pipe(brotli).pipe(res)
+    brotli.pipe(gzip).pipe(res)
```

Comment:
The order of the encoding was wrong

Reply:
this partially undoes c415fbbb59e2b898c5db6a681265cf3da865d02c, I'm not sure it's right. It should be gzipped first and then brotli'd?

Reply:
Accordingly to RFC-9110:
>If one or more encodings have been applied to a representation, the sender that applied the encodings MUST generate a Content-Encoding header field that lists the content codings in the order in which they were used.

At least, in my interpretation, the leftmost is the last applied encoding.
Using `deflate, gzip` as an example, to encode it we first
1. Encode it to `gzip`
2. Encode it to `deflate`

Then to decode it we should do it the other way around, first `gzip`, later `deflate`.
The tests where highlighting this as the header for `gzip` has often wrong when `fetch` was trying to decompress it

Reply:
I believe it's the rightmost which is the last applied encoding, it's in the order the sender compressed it not the order used to decompress it by the client.

> MUST generate a Content-Encoding header field that lists the content codings in the order in which they *were* used

"were" being past tense, this is not in reference to the client who hasn't yet received the content but the server which would be generating the header. And I assume this would be left to right. So your example, ``deflate, gzip`` should be, for the server,
1. Encode it to ``deflate``
2. Encode it to ``gzip``

I've looked around a bit and besides the RFC which is maybe a little unclear, the phrasing everywhere I can see that this is explained (MDN, various tutorials) it's said that it's meant to be the order of the encoding, not the intended decoding order.